tools/firmware.toit

```diff
@@ -732,7 +732,11 @@ extract-host parsed/cli.Parsed envelope/Envelope --config-encoded/ByteArray:
   part-start = bits.size
   // Add the size to have a similar layout to the ESP32 binary.
   bits.little-endian.write-uint32 config-encoded.size
-  bits.write config-encoded
+  // Pad the config to 4 KB. This makes it less likely that the header (which includes
+  // the config-part size) changes for different configurations.
+  // We use 'N', since that's a NOP in UBJSON.
+  padded := pad config-encoded (4 * 1024) --pad-value='N'
```

Comment:
I don't think this is good enough on second thought. We need to reserve space even if config-encoded is zero length.